storage/innobase/include/sync0policy.h

```diff
@@ -52,7 +52,7 @@ class MutexDebug {
 			m_line(),
 			m_thread_id(os_thread_id_t(ULINT_UNDEFINED))
 		{
-			/* No op */
+			m_thread_id_mutex.init();
```

Comment:
Please move this to ```m_context::init()``` (it has to be created) and then call it from ```MutexDebug::init()``` instead of ```m_context::release()```. Still, we need to keep initialisation done by ```m_context::release()``` either by copying or by calling it from ```m_context::init()```.

Reply:
...or initialise directly from ```MutexDebug::init()```